servant-client-core/src/Servant/Client/Core/HasClient.hs

```diff
@@ -7,6 +8,7 @@
 {-# LANGUAGE PolyKinds             #-}
 {-# LANGUAGE RankNTypes            #-}
 {-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeApplications      #-}
```

Comment:
This requires `GHC >= 8.0` which would mean we'd have to adjust cabal bounds; or drop the use of `TypeApplications`

Reply:
Though `sop-core` makes heavy use of `TypeApplications` so perhaps we are past this anyway...

Reply:
@fisx

Reply:
yes, sop-core, plus I really don't want to have to write:

```haskell
   WithStatus (Foo 3 True) :: WithStatus 201 (Foo Bool)
```

instead of

```haskell
   WithStatus @201 (Foo 3 True)
```

I just tried to make that change, but now I say we drop support for ghc <8.  Any objections?